arrays/arrays.c

```diff
@@ -19,45 +19,56 @@ typedef struct Array {
 /*****
  * Allocate memory for a new array
  *****/
-Array *create_array (int capacity) {
+Array *create_array(int capacity)
+{
   // Allocate memory for the Array struct
-
+  Array *newarray = malloc(sizeof(Array));
   // Set initial values for capacity and count
-
+  newarray->capacity = capacity;
+  newarray->count = 0;
   // Allocate memory for elements
+  newarray->elements = calloc(capacity, sizeof(char *));
 
+  return newarray;
 }
 
-
 /*****
  * Free memory for an array and all of its stored elements
  *****/
-void destroy_array(Array *arr) {
+void destroy_array(Array *arr)
+{
 
   // Free all elements
-
+  free(arr->elements);
   // Free array
-
+  free(arr);
 }
 
 /*****
  * Create a new elements array with double capacity and copy elements
  * from old to new
  *****/
-void resize_array(Array *arr) {
+void *resize_array(Array *arr)
+{
 
   // Create a new element storage with double capacity
-
+  int capacity = arr->capacity;
+  char **newelements = calloc(capacity * 2, sizeof(char *));
   // Copy elements into the new storage
-
+  for (int i = 0; i < capacity; i++)
+  {
+    newelements[i] = arr->elements[i];
+    if (newarray->elements[i] != NULL)
+    {
+      newarray->count++;
```

Comment:
Don't need this either, there doesn't need to be any "newarray" here. You're just updating the value of arr->elements in the "old" array. The one that is passed into the function. So the count will stay the same because the number of elements in the array stays the same, you just add more space (i.e. capacity).